tests/Unit/CartTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+use PHPUnit\Framework\TestCase;
+use App\Cart;
+
+class CartTest extends TestCase
+{
+    /**
+     * If item is added to the cart
+     *
+     * @covers Cart::addItem
+     */
+    public function testAddItem()
+    {
+        $cart = new Cart();
+        $cart->addItem('foo');
+        $this->assertEquals(['foo'], $cart->getItems());
+        $this->assertNotEquals(['bar'], $cart->getItems());
+    }
+
+    /**
+     * Remove an item from the cart
+     *
+     * @covers Cart::removeItem
+     */
+    public function testRemoveItem()
+    {
+        $cart = new Cart(['foo']);
+        $cart->removeItem('foo');
+        $this->assertEquals([], $cart->getItems());
+
+        $cart = new Cart(['foo', 'bar', 'horse', 3]);
+        $cart->removeItem(3);
+        $this->assertEquals(['foo', 'bar', 'horse'], $cart->getItems());
+
+        $cart = new Cart(['foo', 'bar', 'horse', 'bar', 'horse', 3]);
+        $cart->removeItem('foo');
+        $this->assertEquals(['bar', 'horse', 'bar', 'horse', 3], $cart->getItems());
+    }
+
+    /**
+     * Get the list of items in the cart
+     *
+     * @covers Cart::getItems
+     */
+    public function testGetItems()
+    {
+        $cart = new Cart();
+        $this->assertEquals([], $cart->getItems());
+
+        $cart = new Cart([1, 2, 3]);
+        $this->assertEquals([1, 2, 3], $cart->getItems());
+    }
+
+    /**
+     * Get the list of items, as a number, in the cart
+     *
+     * @covers Cart::getItems
+     * @depends testGetItems
+     */
+    public function testGetItemsCount()
+    {
+        $cart = new Cart([0, 1, 2, 3]);
+        $this->assertEquals(4, $cart->getItemsCount());
+    }
+
+    public function testUpdateQuantity()
+    {
+        $cart = new Cart(['foo']);
+        $this->assertEquals(['foo'], $cart->getItems());
+        $cart->updateQuantity('foo', 6);
+        $this->assertEquals(['foo', 'foo', 'foo', 'foo', 'foo', 'foo', 'foo'], $cart->getItems());
```

Comment:
For the `foo` example, I was thinking of that as an insertion because of this line:
```$cart->updateQuantity('foo', 6);```
... as in, 0 through 6, which would give 7.

With the cat and dog example, I see that we're literally just keeping one `cat` and tossing out all the other `cat` items.